hack/ci/run-default-application-e2e-test.sh

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+### This script is used as a postsubmit job and updates the dev master
+### cluster after every commit to main.
+
+set -euo pipefail
+
+cd $(dirname $0)/../..
+source hack/lib.sh
+
+TEST_NAME="Pre-warm Go build cache"
+echodate "Attempting to pre-warm Go build cache"
+
+APPLICATION_INSTALLATION_NAME=""
+APPLICATION_NAME=""
+APPLICATION_NAMESPACE=""
+APP_LABEL_KEY=""
+NAMES=""
+
+# Parse arguments
+while [[ "$#" -gt 0 ]]; do
+  case "$1" in
+  --application-installation-name)
+    APPLICATION_INSTALLATION_NAME="$2"
+    shift 2
+    ;;
+  --application-name)
+    APPLICATION_NAME="$2"
+    shift 2
+    ;;
+  --application-namespace)
+    APPLICATION_NAMESPACE="$2"
+    shift 2
+    ;;
+  --app-label-key)
+    APP_LABEL_KEY="$2"
+    shift 2
+    ;;
+  --names)
+    NAMES="$2"
+    shift 2
+    ;;
+  *)
+    echo "Unknown parameter: $1"
+    exit 1
+    ;;
+  esac
+done
```

Comment:
I think the script will rarely be run by humans, maybe we can skip the entire flag parsing and just use env variables?